CHANGELOG.md

```diff
@@ -4,6 +4,10 @@ This changelog keeps track of all changes to the Packs SDK. We follow convention
 
 ## [Unreleased]
 
+### Changed
+
+- Improve types for testing
```

Comment:
The `returns` --> `resolves` thing seems like the most significant change, would call that out. Will external developers who have written unittests also need to update all usage like you did in the `packs` repo? If so I think we need to pre-announce this and explain how to update things. Clearly what was there before was broken so it's a good change but if it's not backwards-compatible we need to give a heads-up.

Reply:
Yeah, if external developers wrote unit tests that didn't correctly return a promise (but instead returned a value), they'd need to update their tests to also use `resolves` to pass type checks. Functionally these tests still work because our code must in general use `await` rather than attempting to do promise chains with `.then()` - the latter would have caused RTEs otherwise.

@ekoleda-codaio can you help with the necessary announcement for this change?

Reply:
@ekoleda-codaio I think the only developer-facing change to announce would be that if somebody previously wrote tests with `newMockExecutionContext()` and they used `context.fetcher.fetch.returns(...)` they need to update it to use `context.fetcher.fetch.resolves(...)`. This should always have been the case but the old behavior was inaccurate considering these are async methods.

Reply:
Thanks for clarifying, I missed this callout. Since tests could only be written with the CLI, and SDK version upgrades are manual and optional for the CLI (compared to the Pack Studio), I think it's fine to release this without an announcement or warning. I also suspect that the number of developers writing tests period is very small, so I expect the blast radius here to be incredibly small. A clear note in the Changelog is likely sufficient, but I can also write up a short community post once the version goes live.

Reply:
Ok that's fine, let's just go with a more thorough changelog entry.

Reply:
Ok, updated the changelog.